hypershift-operator/controllers/hostedcluster/internal/platform/platform.go

```diff
@@ -47,6 +49,8 @@ type Platform interface {
 	ReconcileSecretEncryption(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN,
 		hcluster *hyperv1.HostedCluster,
 		controlPlaneNamespace string) error
+
+	AppendPlatformSpecificPolicyRules(rules []rbacv1.PolicyRule) []rbacv1.PolicyRule
```

Comment:
Please, let's add a comment providing context on how this func should be implemented in the line of the ones for the other interface func.

Reply:
Maybe we want to start with just allowing to return additional rules? In theory the current signature allows more dynamic introspection and addition, but not sure if it is needed, and people can just return nil instead of having to return what was passed in.

Reply:
Changed it, please see https://github.com/openshift/hypershift/pull/779/commits/22e8c33a48c3a7c124821d0fadad6521b3b8cc78